### PR TITLE
triple2grid

### PR DESCRIPTION
**Pull request by abb58**
triple2grid and grid2triple

### CODE REVIEW

File: src/geocat/comp/__init__.py [hunk 1]
```diff
@@ -1105,4 +1105,248 @@ def rcm2points(lat2d, lon2d, fi, lat1dPoints, lon1dPoints, opt=0, msg=None, meta
...
+    if isinstance(ygrid, xr.DataArray):
+        ygrid = ygrid.values
+
+    if isinstance(data, np.ndarray):
+        fo = _ncomp._triple2grid(x, y, data, xgrid, ygrid, options, msg)
+    else:
+        raise TypeError
+
+    if meta and isinstance(input, xr.DataArray):
+        pass	 # TODO: Retaining possible metadata might be revised in the future
```
Comment (clyne) [inline src/geocat/comp/__init__.py]: Rather than silently ignoring this option it would probably be better to throw an error, or remove it. What do others think?

File: src/geocat/comp/__init__.py [hunk 2]
```diff
@@ -1105,4 +1105,248 @@ def rcm2points(lat2d, lon2d, fi, lat1dPoints, lon1dPoints, opt=0, msg=None, meta
...
+    if isinstance(z, xr.DataArray):
+        z = z.values
+
+    if isinstance(z, np.ndarray):
+        fo = _ncomp._grid2triple(x, y, z, msg)
+    else:
+        raise TypeError
+
+    if meta and isinstance(input, xr.DataArray):
+        pass	 # TODO: Retaining possible metadata might be revised in the future
```
Comment (clyne) [inline src/geocat/comp/__init__.py]: Same comment here wrt silently ignoring this option.

File: src/geocat/comp/__init__.py (hunk 1 above)
Comment (abb58) [reply src/geocat/comp/__init__.py]: It would be logical to throw an exception to have an informed feedback that NCL `_wrap()` functions are not yet supported.

Comment (abb58) [reply src/geocat/comp/__init__.py]: https://github.com/NCAR/geocat-comp/pull/53/commits/0d8d4f53013fd903ee03df52ace649a58ea354fa

File: src/geocat/comp/__init__.py (hunk 2 above)
Comment (abb58) [reply src/geocat/comp/__init__.py]: https://github.com/NCAR/geocat-comp/pull/53/commits/0d8d4f53013fd903ee03df52ace649a58ea354fa

File: src/geocat/comp/__init__.py [hunk 3]
```diff
@@ -1105,4 +1110,248 @@ def rcm2points(lat2d, lon2d, fi, lat1dPoints, lon1dPoints, opt=0, msg=None, meta
...
+        data = data.values
+    if isinstance(xgrid, xr.DataArray):
+        xgrid = xgrid.values
+    if isinstance(ygrid, xr.DataArray):
+        ygrid = ygrid.values
+
+    if isinstance(data, np.ndarray):
+        fo = _ncomp._triple2grid(x, y, data, xgrid, ygrid, options, msg)
+    else:
+        raise TypeError
```
Comment (khallock) [inline src/geocat/comp/__init__.py]: We should probably add an actual error message in addition to throwing an exception, maybe something like:
```
raise TypeError("ERROR: triple2grid: unsupported input type {} for 'data' argument.".format(type(data)))

File: src/geocat/comp/__init__.py [hunk 4]
```diff
@@ -1105,4 +1110,248 @@ def rcm2points(lat2d, lon2d, fi, lat1dPoints, lon1dPoints, opt=0, msg=None, meta
...
+        x = x.values
+    if isinstance(y, xr.DataArray):
+        y = y.values
+    if isinstance(z, xr.DataArray):
+        z = z.values
+
+    if isinstance(z, np.ndarray):
+        fo = _ncomp._grid2triple(x, y, z, msg)
+    else:
+        raise TypeError
```
Comment (khallock) [inline src/geocat/comp/__init__.py]: Same as above, we should consider adding an error message to go with the exception.

File: test/test_dpres_plevel.py [hunk 5]
```diff
@@ -25,8 +25,3 @@ def test_rcm2points_float64(self):
         result_dp = geocat.comp.dpres_plevel(plev, psfc)
         np.testing.assert_array_equal(expected_dp, result_dp.values)
 
-class Test_dpres_plevel_float32(ut.TestCase):
-    def test_dpres_plevel_float32(self):
-        plev_asfloat32 = plev.astype(np.float32)
-        result_dp = geocat.comp.dpres_plevel(plev_asfloat32, psfc)
-        np.testing.assert_array_equal(expected_dp, result_dp.values)
```
Comment (khallock) [inline test/test_dpres_plevel.py]: Was this test removed intentionally? This test is for `dpres_plevel` and appears to be unrelated to the `triple2grid` PR.

Comment (khallock) [reply test/test_dpres_plevel.py]: I'm going to create a new PR that re-enables this test so we can diagnose the issue with `dpres_plevel` without holding up this PR.
